libbeat/common/file/file_windows.go

```diff
@@ -107,3 +121,28 @@ func ReadOpen(path string) (*os.File, error) {
 
 	return os.NewFile(uintptr(handle), path), nil
 }
+
+// FileRemoved checks wheter the file held by f is removed.
```

Comment:
This could need a bit more comments I would say on why `GetFileInformationByHandleEx` needs to be used here. Basically what you describe in the PR.